### PR TITLE
Fix many mobile issues with dartdoc

### LINKED ISSUE #908: On mobile, the nav menu doesn't show up on the index page
It does show up for the other pages, like the library pages. By nav menu, I mean the hamburger menu icon.

### PR DESCRIPTION
**Pull request by jcollins-g**
Fixes #908, #1048, #1348, and #1469.  (Some minor bits of these were obsolete before this PR, so technically this doesn't fix everything listed.  But I believe this cleans up enough to address the main points of these bugs and will address other mobile issues separately.)

This PR adds a mobile-only search box to the hamburger button.  Also, this puts the navigation information usually in the title bar in there directly below the search, fixing a host of problems where you can't really get around the API documentation in mobile.

Screenshots for what flutter docs hamburger button menu now looks like:

screenshot from 2018-12-17 16-01-38
screenshot from 2018-12-17 16-03-22

### CODE REVIEW

**Comment by jcollins-g** [comment]
The mobile support and the small screen support is basically the same thing in dartdoc.  So the selectors are on screen size (width, to be specific).  I've rationalized them a bit too in this PR so it is a bit more consistent.

**Comment by pq** [comment]
Ah.  That makes sense.  Thanks for the clarification!

File: lib/resources/styles.css [hunk 1]
```diff
@@ -110,12 +115,18 @@ nav.navbar {
...
-@media (max-width: 767px) {
+@media (max-width: 768px) {
   .hidden-xs {
     display: none!important;
   }
 }
 
+@media (min-width: 769px) {
+  .hidden-l {
+    display: none!important;
```
Comment (devoncarew) [inline lib/resources/styles.css]: there should be a space before `!important`

File: lib/resources/styles.css [hunk 2]
```diff
@@ -110,12 +115,18 @@ nav.navbar {
   border: 0;
 }
 
-@media (max-width: 767px) {
+@media (max-width: 768px) {
   .hidden-xs {
     display: none!important;
```
Comment (devoncarew) [inline lib/resources/styles.css]: not part of this PR, but this needs a space before `!important`

File: lib/resources/styles.css (hunk 1 above)
Comment (jcollins-g) [reply lib/resources/styles.css]: done

File: lib/resources/styles.css (hunk 2 above)
Comment (jcollins-g) [reply lib/resources/styles.css]: done
